lib/components/narrative/tabbed-itineraries.js

```diff
@@ -44,17 +48,43 @@ class TabButton extends Component {
       itinerary,
       true
     )
-    // TODO: support non-USD
     const transitFare =
-      (transitFares?.[defaultFareKey] || transitFares.regular)?.transitFare || 0
+      (transitFares?.[defaultFareKey] || transitFares.regular)?.transitFare ||
+      null
+    const fareCurrency = itinerary.fare?.fare?.regular?.currency?.currencyCode
     const minTotalFare = minTNCFare * 100 + transitFare
     const startTime = itinerary.startTime + timezoneOffset
     const endTime = itinerary.endTime + timezoneOffset
     const mustCallAhead = itinerary.legs.some(
       coreUtils.itinerary.isReservationRequired
     )
+    const transitFareNotProvided = itinerary.legs.some(
+      (leg) => isTransit(leg.mode) && transitFare == null
+    )
 
     if (isActive) classNames.push('selected')
+    const newLocal = minTotalFare > 0 && (
+      <>
+        <FormattedMessage
+          id="components.TabbedItineraries.fareCost"
+          values={{
+            hasMaxFare: maxTNCFare && maxTNCFare > minTNCFare,
+            minTotalFare: (
+              <FormattedNumber
+                // Currency from itinerary fare or from config.
+                currency={fareCurrency || currency}
```

Comment:
Use `fareCurrency` and append `|| currency` to its definition.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/565/commits/d9c5b8309c89e75580bfaac4e812dbd10f4bda29